src/agent0/ethpy/hyperdrive/addresses.py

```diff
@@ -151,7 +151,22 @@ def get_hyperdrive_addresses_from_registry(
     return addresses
 
 
-def _generate_name_for_hyperdrive(hyperdrive_address: ChecksumAddress, web3: Web3) -> str:
+def generate_name_for_hyperdrive(hyperdrive_address: ChecksumAddress, web3: Web3) -> str:
+    """Generates a name for a given hyperdrive address. The address generated is of the form
+    <vault_shares_token_symbol>_<position_duration>_day.
+
+    Arguments
+    ---------
+    hyperdrive_address: ChecksumAddress
+        The address of a hyperdrive pool.
+    web3: Web3
+        The instantiated Web3 instance that's connect to a chain to use.
+
+    Returns
+    -------
+    str
+        The generated name of the hyperdrive pool.
+    """
     # TODO ideally we would use the HyperdriveReadInterface here, but that creates a circular
```

Comment:
what if this whole fn existed in the interface? Instead of it living here & using the interface. Does that resolve the circular dependency?

Reply:
Unfortunately, this function is used in `get_hyperdrive_addresses_from_registry`, which doesn't build the interface, and is accessed by passing in a registry address before anything gets initialized. I'll keep this here for the time being.